libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -734,18 +734,27 @@ abstract class AbstractAndroidProjectHandler(private val kotlinConfigurationTool
         val project = kotlinAndroidTarget.project
         val ext = project.extensions.getByName("android") as BaseExtension
 
-        ext.sourceSets.all { sourceSet ->
-            logger.kotlinDebug("Creating KotlinBaseSourceSet for source set $sourceSet")
-            val kotlinSourceSet = project.kotlinExtension.sourceSets.maybeCreate(
-                lowerCamelCaseName(kotlinAndroidTarget.disambiguationClassifier, sourceSet.name)
-            ).apply {
-                kotlin.srcDir(project.file(project.file("src/${sourceSet.name}/kotlin")))
-                kotlin.srcDirs(sourceSet.java.srcDirs)
+        fun kotlinSourceSetName(sourceSet: AndroidSourceSet): String {
+            return lowerCamelCaseName(kotlinAndroidTarget.disambiguationClassifier, sourceSet.name)
+        }
+
+        val kotlinSourceSetNames = ext.sourceSets.map(::kotlinSourceSetName)
+
+        ext.sourceSets.all { androidSourceSet ->
+            logger.kotlinDebug("Creating KotlinBaseSourceSet for source set $androidSourceSet")
+            val kotlinSourceSet = project.kotlinExtension.sourceSets.maybeCreate(kotlinSourceSetName(androidSourceSet)).apply {
+                /* Avoiding conflicts like the "androidTest" kotlin source set vs "androidTest" android source set. */
+                if (androidSourceSet.name !in kotlinSourceSetNames) {
+                    kotlin.srcDir(project.file(project.file("src/${androidSourceSet.name}/kotlin")))
+                    kotlin.srcDirs(androidSourceSet.java.srcDirs)
```

Comment:
Consider a single-platform project, where the target's `disambiguationClassifier` is null and `kotlinSourceSetName` will thus be always equal to the Android source set name. The condition `androidSourceSet.name !in kotlinSourceSetNames` will always be false, and, in particular, this line won't be executed for any source set:
```
kotlin.srcDirs(androidSourceSet.java.srcDirs)
```
I haven't checked this locally, but, according to my understanding, this will break the ability for a user to place Kotlin sources under the Android source set's `src/*/java` srcDir, which is commonplace in existing projects.

If I understand the intention correctly, this check should only be applied when the Kotlin source set names differ from the Android source set names.

Reply:
> this will break the ability for a user to place Kotlin sources under the Android source set's src/*/java srcDir

Sorry, my mistake, this is not the case, as the Java sources are directly added to the task [here](https://github.com/JetBrains/kotlin/blob/master/libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt#L996).

Reply:
Oh yes! You are right: I did not think about the fact, that this `disambiguationClassifier` can be null! But I think simply checking for this case as well would do the trick there?

Other than that: 
Yes, this will not register the 
`src/androidTest/java` (android source dir) into the `androidAndroidTest` kotlin src dir, but this is a good thing IMO ☺️ 👍

Reply:
I updated the fix now. With this, all ambiguous java source folders are also disabled 
- `src/androidTest/kotlin` ✅ 
- `src/androidTest/java` :x:

<img width="267" alt="Bildschirmfoto 2019-11-26 um 19 19 08" src="https://user-images.githubusercontent.com/34319766/69660938-abf4e380-1081-11ea-8aeb-58e678155c84.png">

Sample project can be viewed at https://github.com/sellmair/mpp-playground/tree/connectedAndroidTest-not-executed-fix

I think supporting the  `src/androidTest/java` folder is just confusing since it would be actual instrumented tests!

Reply:
I also tested a single platform Android module. Seems to be working as well:
<img width="363" alt="Bildschirmfoto 2019-11-26 um 21 53 59" src="https://user-images.githubusercontent.com/34319766/69671947-4e6b9180-1097-11ea-8a5f-88dd64c79347.png">

Reply:
I'll try to push Gradle Integration tests in the following days 👍